fire/core.py

```diff
@@ -785,15 +789,27 @@ def _ParseKeywordArgs(args, fn_spec):
       continue
 
     arg_consumed = False
-    if argument.startswith('--'):
+    if _IsFlag(argument):
       # This is a named argument; get its value from this arg or the next.
       got_argument = False
 
-      keyword = argument[2:]
+      keyword = ''
+      if _IsSingleCharFlag(argument):
+        keychar = argument[1]
+        potential_args = [arg for arg in fn_args if arg[0] == keychar]
+        if len(potential_args) == 1:
+          keyword = potential_args[0]
+        elif len(potential_args) > 1:
```

Comment:
we also need to handle the len(potential_args) == 0 case.

Reply:
Should we also throw an error in this case?  My original assumption here was that if the flag doesn't match, we would assume it was an value for another flag.  Now that I think about it though this could cause confusing behavior.

Reply:
Yeah I think we should throw an error in this case.
If the argument looks like a flag (_IsFlag), it shouldn't be acceptable as a value.
[And I think we'll also broaden what _IsFlag accepts too in a separate change.]

Reply:
Ok that sounds good to me, I'll add a branch and a test case for it